authz/providers/azure/rbac/checkaccessreqhelper.go

```diff
@@ -198,34 +194,237 @@ func getActionName(verb string) string {
+							finalFilteredOperations = initializeMapForGroupAndResource(finalFilteredOperations, group, resourceName)
+							finalFilteredOperations[group][resourceName] = verbValues
+							break
+						}
+					}
+				}
+			}
+		} else {
+			// both cluster scoped and namespace scoped resource list
+			finalFilteredOperations = filteredOperations
+		}
+
+		klog.V(7).Infof("List of filtered operations: %s", finalFilteredOperations)
+
+		// create list of Data Actions
+		authInfoList, err = createAuthorizationActionInfoList(finalFilteredOperations, subRevReq.ResourceAttributes.Verb)
+		if err != nil {
+			return nil, err
+		}
+
+	} else {
+		/*
+			   This sections handles the following scenarios:
+
+				| Scenario        | Namespace is empty (Cluster scope call)          | Namespace is not empty (NS scope)                    |
+				------------------ --------------------------------------------------  -----------------------------------------------------
+				| Verb-*, Group-* | Resource under all apigroups and with all verbs  | Resource under specifed apigroups and with all verbs |
+								  | at clusterscope                                  | at ns scope                                          |
+
+				| Verb - *        | Resource under specifed apigroups and with all   | Resource under specifed apigroups and with all verbs |
+								  | verbs at cluster scope                           | at ns scope                                          |
+
+				| Group - *       | Resource under all apigroups with specified verb | Resource under all apigroups with specified verb     |
+				|                 | at cluster scope                                 |  at ns scope                                         |
+		*/
+
+		filteredOperations := azureutils.NewOperationsMap()
+		if subRevReq.ResourceAttributes.Group == "*" {
+			// #1 and #3
+			for group, resMap := range operationsMap {
+				if verbMap, found := resMap[subRevReq.ResourceAttributes.Resource]; found {
+					filteredOperations = initializeMapForGroupAndResource(filteredOperations, group, subRevReq.ResourceAttributes.Resource)
+					filteredOperations[group][subRevReq.ResourceAttributes.Resource] = verbMap
+				}
+			}
+		} else { // #2
+			group := "v1" // core api group key
+			if subRevReq.ResourceAttributes.Group != "" {
+				group = subRevReq.ResourceAttributes.Group
+			}
+
+			if resMap, found := operationsMap[group]; found {
+				if verbMap, found := resMap[subRevReq.ResourceAttributes.Resource]; found {
+					filteredOperations = initializeMapForGroupAndResource(filteredOperations, group, subRevReq.ResourceAttributes.Resource)
+					filteredOperations[group][subRevReq.ResourceAttributes.Resource] = verbMap
+				}
+			} else {
+				return nil, errors.Errorf("No resources found for group %s and resource %s", subRevReq.ResourceAttributes.Group, subRevReq.ResourceAttributes.Resource)
+			}
+
+		}
+
+		klog.V(7).Infof("List of filtered operations: %s", filteredOperations)
```

Comment:
do we need to remove cluster scope resources if ns is not empty for specific resource case?